recipes/gperf/all/test_package/conanfile.py

```diff
@@ -0,0 +1,8 @@
+from conans import ConanFile, tools
+
+
+class TestPackageConan(ConanFile):
+
+    def test(self):
+        if not tools.cross_building(self.settings):
+            self.run("gperf --version")
```

Comment:
This test might pick up a globally installed gperf.
Maybe add a environment variable `GPERF` that contains the path to gperf?

I also think the test is too simple.
I found some simple examples at https://www.lrde.epita.fr/~tiger/doc/gnuprog2/Simple-Uses-of-Gperf.html

Reply:
Adding `run_environment=True` should be good enough

This is a standard test for all of our installer packages. Either we should come to the conclusion that we want to have more extensive tests for all installers or this should be fine

Reply:
I guess it's a matter of trust and I don't trust my computer 😃

Reply:
> This test might pick up a globally installed gperf.

This is true for all recipes, even non installer. Maybe the most generic way to test would be to do like [cmake](https://github.com/conan-io/conan-center-index/blob/master/recipes/cmake/all/test_package/conanfile.py)

Reply:
Okay, `run_environment=True` is not the answer here. The `bin` directory is appended in both cases, so it wouldn't prevent when gperf is not found within the Conan directories and picked up from the system

But again, if we consider this a general problem it would apply to most installer packages we have.

The CI system probably doesn't have most installer packages pre-installed so we should catch such errors at the minimum with CI runs.

Does that sound good enough?

Reply:
Good point about non-installer packages.

Reply:
Sounds fine to me, I just want to point at autoconf where I created environment variables for the important scripts.
These are needed to be able to package autoconf, automake and libtool using conan.

But these variables are also handy to test the scripts because the msys2 package has libtool already installed.